### PR TITLE
A new rule as a part of CIS profile for SLE 12/15 - all_apparmor_prof…

### PR DESCRIPTION
**Pull request by rumch-se**
…iles_enforced

#### Description:

- _This rule covers CIS recommendation Ensure all AppArmor Profiles are enforcing (Automated) _

#### Rationale:

- The existing CIS profile for SLE 12/15 does not cover this recommendation. This recommendation can be extended by other Linux vendors.

### CODE REVIEW

**Comment by dodys** [comment]
This right now is conflicting with PR #9985 and we should try to get to a common agreement

File: controls/cis_sle12.yml [hunk 1]
```diff
@@ -642,7 +644,7 @@ controls:
     status: automated
     rules:
       - service_rpcbind_disabled
-      - package_rpcbind_removed
+      - package_rcpbind_removed
```
Comment (dodys) [inline controls/cis_sle12.yml:647]: this shouldn't be part of this PR, please fix it in another PR.

File: controls/cis_sle15.yml [hunk 2]
```diff
@@ -623,7 +625,7 @@ controls:
     status: automated
     rules:
       - service_rpcbind_disabled
-      - package_rpcbind_removed
+      - package_rcpbind_removed
```
Comment (dodys) [inline controls/cis_sle15.yml:628]: same as previous comment

**Review by dodys — CHANGES_REQUESTED** [review]
just some small changes
thanks for providing this.

File: linux_os/guide/system/apparmor/all_apparmor_profiles_enforced/oval/shared.xml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  <!-- Define "3rd" test for processes. -->
+  <linux:apparmorstatus_test id="test_processes_are_confined" version="1"
+    check="all" check_existence="at_least_one_exists"
+    comment="Zero processes are unconfined" >
+    <linux:object object_ref="object_apparmor_profiles_existing" /> 
+    <linux:state state_ref="no_unconfined_processes" />
+  </linux:apparmorstatus_test>
+
+  <linux:apparmorstatus_object id="object_apparmor_profiles_existing" version="1">
+  </linux:apparmorstatus_object >
```
Comment (dodys) [inline linux_os/guide/system/apparmor/all_apparmor_profiles_enforced/oval/shared.xml:36]: you can consolidate this in one line:
`<linux:apparmorstatus_object id="object_apparmor_profiles_existing" version="1" />`

**Comment by dodys** [comment]
fyi, since we don't have a SUSE maintainer and this rule also is needed for Ubuntu, I've assigned it to me.

**Comment by dodys** [comment]
This rule is not buildable in Ubuntu (and Debian), as apparmor schema was only added in OVAL Format 5.11.2 and Ubuntu now only has OVAL Format 5.11.1.

@rumch-se can you confirm that this is also the case for OpenSUSE, as the logs are showing that it doesn't recognize the apparmor schema.

**Comment by rumch-se** [comment]
Hello @dodys 

I have done changed proposed.
Have a nice day
Rumen

File: controls/cis_sle15.yml [hunk 4]
```diff
@@ -625,7 +625,7 @@ controls:
     status: automated
     rules:
       - service_rpcbind_disabled
-      - package_rcpbind_removed
+      - package_rpcbind_removed
```
Comment (dodys) [inline controls/cis_sle15.yml:628]: @rumch-se this is still present, please remove it from this PR and send it in another PR.

File: controls/cis_sle12.yml [hunk 5]
```diff
@@ -644,7 +644,7 @@ controls:
     status: automated
     rules:
       - service_rpcbind_disabled
-      - package_rcpbind_removed
+      - package_rpcbind_removed
```
Comment (dodys) [inline controls/cis_sle12.yml:647]: @rumch-se this is still present, please remove it from this PR and send it in another PR.

**Review by dodys — CHANGES_REQUESTED** [review]

File: controls/cis_sle12.yml (hunk 5 above)
Comment (rumch-se) [reply controls/cis_sle12.yml:647]: Hello @dodys

I can remove package_rpcbind_removed, but its points to the rule
/linux_os/guide/services/nfs_and_rpc/disabling_nfs/disabling_nfs_services/package_rpcbind_removed

or ... I have to remove something else ?
Please describe
 
Have a nice day
Rumen

File: controls/cis_sle15.yml (hunk 4 above)
Comment (rumch-se) [reply controls/cis_sle15.yml:628]: Hello @dodys

I can remove package_rpcbind_removed, but its points to the rule
/linux_os/guide/services/nfs_and_rpc/disabling_nfs/disabling_nfs_services/package_rpcbind_removed

or ... I have to remove something else ?
Please describe

Have a nice day
Rumen

File: controls/cis_sle12.yml (hunk 5 above)
Comment (dodys) [reply controls/cis_sle12.yml:647]: let's leave it as-is

**Comment by dodys** [comment]
@marcusburghardt @mildas this has the same point as asked in PR #10064 
it won't build on environments with OVAL version < 5.11.2

**Review by dodys — APPROVED** [review]

**Comment by dodys** [comment]
Hi @rumch-se, could you please rebase this PR and remove the commit to add the OVAL. Since your OVAL is dependent of OVAL format 5.11.2 and most of the CI only has OVAL format 5.11.1, we won't be able to merge it as this will break the CI.
I will later add an SCE audit script for ubuntu.
@teacup-on-rockingchair fyi ^ 
let us know if you have any doubts.

**Comment by rumch-se** [comment]
Hello @dodys 
I did rebase 1st , and after I removed the OVAL part
Have a nice day
Rumen

**Review by dodys — APPROVED** [review]
LGTM, thanks!

@marcusburghardt there's one test failing that I think has nothing to do with this PR.

**Comment by dodys** [comment]
@marcusburghardt @Mab879 can we get this merged? I'm not sure how I overwrite codeowners

**Comment by dodys** [comment]
@marcusburghardt ping

File: linux_os/guide/system/apparmor/all_apparmor_profiles_enforced/ansible/shared.yml [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# platform = multi_platform_sle
+# reboot = false
+# strategy = restrict
+# complexity = low
+# disruption = low
+
+- name: Ensure all AppArmor Profiles are reloaded
+  ansible.builtin.shell: apparmor_parser -q -r /etc/apparmor.d/
```
Comment (marcusburghardt) [inline linux_os/guide/system/apparmor/all_apparmor_profiles_enforced/ansible/shared.yml:8]: Please, use the `command` module instead of `shell` when the `command` is enough.

File: linux_os/guide/system/apparmor/all_apparmor_profiles_enforced/ansible/shared.yml [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+- name:  Ensure all AppArmor Profiles are enforcing
+  ansible.builtin.shell: aa-enforce /etc/apparmor.d/*
```
Comment (marcusburghardt) [inline linux_os/guide/system/apparmor/all_apparmor_profiles_enforced/ansible/shared.yml:11]: `command` module is enough here.

File: linux_os/guide/system/apparmor/all_apparmor_profiles_enforced/ansible/shared.yml [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+- name: Collect unconfined processes 
+  ansible.builtin.shell: aa-unconfined
```
Comment (marcusburghardt) [inline linux_os/guide/system/apparmor/all_apparmor_profiles_enforced/ansible/shared.yml:14]: `command` module is enough here.

File: linux_os/guide/system/apparmor/all_apparmor_profiles_enforced/ansible/shared.yml [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+  register: unconfined_processes
+
+- name: Provide details about unconfined processes 
+  ansible.builtin.shell: echo "The process {{ item }} may need to have a profile created or activated for them and then be restarted."
```
Comment (marcusburghardt) [inline linux_os/guide/system/apparmor/all_apparmor_profiles_enforced/ansible/shared.yml:18]: I recommend to use the [assert](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/assert_module.html) module here.

Example:
https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/services/ssh/ssh_server/firewalld_sshd_port_enabled/ansible/shared.yml#L80

**Review by marcusburghardt — CHANGES_REQUESTED** [review]
I have only some comments about the Ansible remediation.
Style Guide: https://complianceascode.readthedocs.io/en/latest/manual/developer/04_style_guide.html#ansible

With these small improvements the PR should be ready to be merged.

**Comment by rumch-se** [comment]
Hello @marcusburghardt 
All proposed changes were done
Have a nice day 
Rumen

**Review by marcusburghardt — APPROVED** [review]
